Services/SystemServer/main.cpp

```diff
@@ -84,6 +85,86 @@ static void parse_boot_mode()
     dbg() << "Booting in " << g_boot_mode << " mode";
 }
 
+static void prepare_devfs()
+{
+    // FIXME: Find a better way to all of this stuff, without hardcoding all of this!
+
+    int rc = mount(-1, "/dev", "dev", 0);
+    if (rc != 0) {
+        ASSERT_NOT_REACHED();
+    }
+    rc = mkdir("/dev/pts", 0755);
+    if (rc != 0) {
+        ASSERT_NOT_REACHED();
+    }
+    rc = mount(-1, "/dev/pts", "devpts", 0);
+    if (rc != 0) {
+        ASSERT_NOT_REACHED();
+    }
+    rc = symlink("/proc/self/fd/0", "/dev/stdin");
+    if (rc < 0) {
+        ASSERT_NOT_REACHED();
+    }
+    rc = symlink("/proc/self/fd/1", "/dev/stdout");
+    if (rc < 0) {
+        ASSERT_NOT_REACHED();
+    }
+    rc = symlink("/proc/self/fd/2", "/dev/stderr");
+    if (rc < 0) {
+        ASSERT_NOT_REACHED();
+    }
+    rc = symlink("/dev/random", "/dev/urandom");
+    if (rc < 0) {
+        ASSERT_NOT_REACHED();
+    }
```

Comment:
Can we avoid doing this in userspace? Can DevFS create this automatically instead?

Reply:
I guess we could create a symlink in the DevFS code. As I said in the comment above, it's up to you :)